log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/rolling/RollingFileManagerTest.java

```diff
@@ -188,4 +193,40 @@ public void testCreateParentDir() {
             manager.close();
         }
     }
+
+    @Test
+    @Issue("https://github.com/apache/logging-log4j2/issues/2592")
+    public void testRolloverOfDeletedFile() throws IOException {
+        final Configuration configuration = new NullConfiguration();
+        final File file = File.createTempFile("testRolloverOfDeletedFile", "log");
+        file.deleteOnExit();
+        final String testContent = "Test";
+        try (final OutputStream os =
+                        new ByteArrayOutputStream(); // use a dummy OutputStream so that the real file can be deleted
+                final RollingFileManager manager = new RollingFileManager(
```

Comment:
Nice test! :100:

By calling the constructor directly, you prevent the registration of `RollingFileManager` in the registry kept by `AbstractManager`. This a very nice approach for tests!